pkg/test/case.go

```diff
@@ -355,7 +344,7 @@ func (t *Case) Run(test *testing.T, tc *report.Testcase) {
 		tc.Assertions += len(testStep.Asserts)
 		tc.Assertions += len(testStep.Errors)
 
-		if errs := runStep(test, t, testStep, ns); len(errs) > 0 {
+		if errs := testStep.Run(test, ns.Name, t.SkipDelete); len(errs) > 0 {
```

Comment:
Not sure it would make things better, currently this is not overridable in a `Step`, this makes sense to me that it is the responsibility of the caller to pass the flag.
I don't have a strong opinion though, if you prefer the field based approach I will use that.
WDYT ?

Reply:
Switched to a struct field.